drivers/timer/stm32_lptim_timer.c

```diff
@@ -227,15 +229,20 @@ void z_clock_set_timeout(s32_t ticks, bool idle)
 	if (next_arr > LPTIM_TIMEBASE) {
 		next_arr = LPTIM_TIMEBASE;
 	}
+	/* The new autoreload value must be 2 cycle after current lptim
+	 * to make sure we don't miss an autoreload interrupt
+	 */
+	else if (next_arr < (lp_time + 2)) {
+		next_arr = lp_time + 2;
+	}
 
-	/* If we are close to the roll over of the ticker counter
-	 * change current tick so it can be compared with buffer.
-	 * If this event got outdated fire interrupt right now,
-	 * else schedule it normally.
+	/* If the autoreload interrupts are too close together,
+	 * it will be impossible to guarantee to set autoreload correctly:
+	 * we must have at least 2 clock cycle to have some times to set
+	 * new autoreload value.
 	 */
-	if (next_arr <= ((lp_time + 1) & LPTIM_TIMEBASE)) {
-		NVIC_SetPendingIRQ(LPTIM1_IRQn);
-		lptim_fired = 1;
+	else if (next_arr < 2) {
+		next_arr = 2;
```

Comment:
BTW, if the guard value is at least 2 lptim clock cycles, could it be a kind of constant _LPTIM_GUARD_VALUE_
for each comparison here and at lines above.